docs/graph.md

```diff
@@ -0,0 +1,714 @@
+# Graphs
+
+!!! danger "Don't use a nail gun unless you need a nail gun"
+    If PydanticAI [agents](agents.md) are a hammer, and [multi-agent workflows](multi-agent-applications.md) are a sledgehammer, then graphs are a nail gun, with flames down the side:
+
+    * sure, nail guns look cooler than hammers
+    * but nail guns take a lot more setup than hammers
+    * and nail guns don't make you a better builder, they make you a builder with a nail gun
+    * Lastly, (and at the risk of torturing this metaphor), if you're a fan of medieval tools like mallets and untyped Python, you probably won't like nail guns or our approach to graphs. (But then again, if you're not a fan of type hints in Python, you've probably already bounced off PydanticAI to use one of the toy agent frameworks — good luck, and feel free to borrow my sledgehammer when you realize you need it)
```

Comment:
1, I'm not sure if the metaphors here are sound (do nail guns really take a lot of setup? lol), and 2, I feel like this warning bubble is perhaps a bit overly strongly stated.

I feel like the current phrasing comes across a bit like you are comparing people who are afraid of the graph-based approach to people that who are afraid of types. And I don't think that's the intent, so I think it makes sense to try to just focus on "if you don't think you need this, you probably don't".

Reply:
we can tone it down a bit, but right now I think this is an okay start.